app/pages/Auction/BidActionPanel/BidNow.js

```diff
@@ -59,11 +63,19 @@ class BidNow extends Component {
   };
 
   sendBid = async () => {
-    const {sendBid} = this.props;
+    const {sendBid, domain} = this.props;
     const {bidAmount, disguiseAmount} = this.state;
     const lockup = Number(disguiseAmount) + Number(bidAmount);
 
     try {
+      // If the name is not already stored in the wallet, rescan the auction
+      // for past bids before proceeding. The wallet will continue to track
+      // newer bids automatically.
+      if (!domain.walletHasName) {
+        this.props.waitForWalletSync();
```

Comment:
Yeah its funky because we want the wallet progress modal to run asynchronously in "parallel" with the actual wallet import-and-rescan process. I played with a few combinations of call sequences and awaits and this was the best UX (modal appears and stays up until rescan is done)

Although looking at it again it looks like an edge case could exist where the progress modal will load, quickly hit 100% and exit before the wallet import-and-rescan has even started, but I haven't seen that in any of my tests.

The sequence is a little different when restoring from seed. This was required because the import seed function actually restarts the wallet service completely, so the progress monitor must be called last:

https://github.com/pinheadmz/bob-wallet/blob/af995a7b1a0a2fd9b00cb791606647f2aa3f5a0f/app/pages/Onboarding/ImportSeedFlow/index.js#L118-L120

Reply:
see comment below